basyx/aas/adapter/xml/xml_deserialization.py

```diff
@@ -1291,88 +1315,54 @@ def read_aas_xml_element(file: IO, construct: XMLConstructables, failsafe: bool
+        XMLConstructables.BASIC_EVENT_ELEMENT: decoder_.construct_basic_event_element,
+        XMLConstructables.BLOB: decoder_.construct_blob,
+        XMLConstructables.CAPABILITY: decoder_.construct_capability,
+        XMLConstructables.ENTITY: decoder_.construct_entity,
+        XMLConstructables.EXTENSION: decoder_.construct_extension,
+        XMLConstructables.FILE: decoder_.construct_file,
+        XMLConstructables.RESOURCE: decoder_.construct_resource,
+        XMLConstructables.MULTI_LANGUAGE_PROPERTY: decoder_.construct_multi_language_property,
+        XMLConstructables.OPERATION: decoder_.construct_operation,
+        XMLConstructables.PROPERTY: decoder_.construct_property,
+        XMLConstructables.RANGE: decoder_.construct_range,
+        XMLConstructables.REFERENCE_ELEMENT: decoder_.construct_reference_element,
+        XMLConstructables.RELATIONSHIP_ELEMENT: decoder_.construct_relationship_element,
+        XMLConstructables.SUBMODEL_ELEMENT_COLLECTION: decoder_.construct_submodel_element_collection,
+        XMLConstructables.SUBMODEL_ELEMENT_LIST: decoder_.construct_submodel_element_list,
+        XMLConstructables.ASSET_ADMINISTRATION_SHELL: decoder_.construct_asset_administration_shell,
+        XMLConstructables.ASSET_INFORMATION: decoder_.construct_asset_information,
+        XMLConstructables.SPECIFIC_ASSET_ID: decoder_.construct_specific_asset_id,
+        XMLConstructables.SUBMODEL: decoder_.construct_submodel,
+        XMLConstructables.VALUE_REFERENCE_PAIR: decoder_.construct_value_reference_pair,
+        XMLConstructables.CONCEPT_DESCRIPTION: decoder_.construct_concept_description,
+        XMLConstructables.MULTI_LANGUAGE_NAME_TYPE: decoder_.construct_multi_language_name_type,
+        XMLConstructables.MULTI_LANGUAGE_TEXT_TYPE: decoder_.construct_multi_language_text_type,
+        XMLConstructables.DEFINITION_TYPE_IEC61360: decoder_.construct_definition_type_iec61360,
+        XMLConstructables.PREFERRED_NAME_TYPE_IEC61360: decoder_.construct_preferred_name_type_iec61360,
+        XMLConstructables.SHORT_NAME_TYPE_IEC61360: decoder_.construct_short_name_type_iec61360,
+        XMLConstructables.EMBEDDED_DATA_SPECIFICATION: decoder_.construct_embedded_data_specification,
+        XMLConstructables.DATA_SPECIFICATION_IEC61360: decoder_.construct_data_specification_iec61360,
+        # the following constructors decide which constructor to call based on the elements tag
+        XMLConstructables.DATA_ELEMENT: decoder_.construct_data_element,
+        XMLConstructables.SUBMODEL_ELEMENT: decoder_.construct_submodel_element,
+        XMLConstructables.DATA_SPECIFICATION_CONTENT: decoder_.construct_data_specification_content,
+        # type aliases
+        XMLConstructables.VALUE_LIST: decoder_.construct_value_list,
+    }
+
+    constructor: Callable[..., object]
+    if construct in type_constructors:
+        constructor = type_constructors[construct]
```

Comment:
This can be shortened to something like
```python
constructor: Optional[Callable[..., object]] = type_constructors.get(construct)
if constructor is None:
    raise ValueError(...)
```

Reply:
Done in https://github.com/eclipse-basyx/basyx-python-sdk/pull/166/commits/dc421bd614def34e805e5f846ac823cdf29230ca